File: chatappBE/common/common-events/src/main/java/com/example/common/integration/presence/PresenceStatus.java
package com.example.common.integration.presence;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum PresenceStatus {

    ONLINE("ONLINE"),
    AWAY("AWAY"),
    OFFLINE("OFFLINE");

    private final String value;

    PresenceStatus(String value) {
        this.value = value;
    }

    @JsonValue
    public String value() {
        return value;
    }

    @JsonCreator
    public static PresenceStatus fromValue(String value) {
        for (PresenceStatus status : values()) {
            if (status.value.equalsIgnoreCase(value)) {
                return status;
            }
        }
        throw new IllegalArgumentException("Unknown PresenceStatus: " + value);
    }
}

File: chatappFE/src/utils/presenceStatus.ts
import type { PresenceStatus } from "../types/presence"

export const PRESENCE_STATUS_ORDER: Record<PresenceStatus, number> = {
  ONLINE: 0,
  AWAY: 1,
  OFFLINE: 2,
}


export const getStatusDotClass = (status: PresenceStatus): string => {
  switch (status) {
    case "ONLINE":
      return "bg-green-500"
    case "AWAY":
      return "bg-amber-400"
    case "OFFLINE":
    default:
      return "bg-gray-500"
  }
}
